ignore vesa modes where either bits_per_pixel or depth is 8 or less.

--- linux-2.6.16.orig/drivers/video/vesafb-tng.c	Tue May  2 20:06:30 2006
+++ linux-2.6.16/drivers/video/vesafb-tng.c	Sat May  6 11:37:59 2006
@@ -780,19 +780,20 @@
 		/* We only want modes that are supported with the currennt
 		 * hardware configuration (D0), color (D3), graphics (D4)
 		 * and that have support for the LFB (D7). */
-		if ((mib->mode_attr & 0x99) == 0x99 &&
-		    mib->bits_per_pixel >= 8) {
-			off++;
-		} else {
-			vbe_modes_cnt--;
-		}
-		mode++;
 		mib->depth = mib->red_len + mib->green_len + mib->blue_len;
 		/* Handle 8bpp modes and modes with broken color component
 		 * lengths. */
 		if (mib->depth == 0 ||
 		    (mib->depth == 24 && mib->bits_per_pixel == 32))
 			mib->depth = mib->bits_per_pixel;
+
+		if ((mib->mode_attr & 0x99) == 0x99 &&
+		    mib->bits_per_pixel > 8 && mib->depth >= 8) {
+			off++;
+		} else {
+			vbe_modes_cnt--;
+		}
+		mode++;
 	}
 
 	return 0;
